service/alerter/service.go

```diff
@@ -204,6 +234,33 @@ func FindOrphanResources(clusterIDs []string, resourceNames []string) []string {
 	return orphanResources
 }
 
+// FindOrphanClusters compares a list of cluster IDs and resource names. It
+// returns the IDs of any cluster created a reasonable amount of time ago without
+// resources associated.
+func FindOrphanClusters(awsConfigs []v1alpha1.AWSConfig, resourceNames []string) []string {
+	resources := map[string]bool{}
+	orphanClusters := []string{}
+
+	for _, resourceName := range resourceNames {
+		resources[resourceName] = true
+	}
+
+	n := time.Now()
+	l := n.Add(-1 * maxMinutesSinceClusterCreation * time.Minute)
+	for _, awsConfig := range awsConfigs {
+		// if the cluster has been recently created maybe the resources are not still there.
+		if awsConfig.CreationTimestamp.After(l) {
```

Comment:
As a random point, it feels like prometheus should take care of this part. If the alerting rule has a long period, for example.

Reply:
I think a longer period wouldn't help here. Imagine that, by the time the rule kicks in, a cluster has been just created, if we don't check the creation timestamp we might think that the cluster is orphaned because its resources are not yet created, while this is totally normal (it takes some time to create the resources after the custom object is detected). 

IMO at some point we need to check the actual cluster creation time and only try to find resources of old enough clusters, this place seems to me like the more appropriate, or maybe I'm not getting right the longer period you suggest?

Reply:
So, my thinking was to expose the metrics directly to Prometheus, and then in the Prometheus alert define how long it should be till the alert is fired. Not certain it's a massive issue, but having the logic here feels slightly off?

Reply:
I reckon the alerts would be specific to the operators? Like, the aws alerts are all for the aws-operator anyway

If you wanna hang out and chat about aws-operator exposing more metrics, I would also love that too :D

Reply:
Indeed, this alert is specific to this operator, maybe could be extended to other cloud operators, uncertain.